validation/jvm/src/test/scala/hmda/validation/engine/ts/TsValidationSpec.scala

```diff
@@ -0,0 +1,29 @@
+package hmda.validation.engine.ts
+
+import hmda.parser.fi.ts.TsGenerators
+import org.scalacheck.Gen
+import org.scalatest.prop.PropertyChecks
+import org.scalatest.{ MustMatchers, PropSpec }
+import scala.concurrent.{ ExecutionContext, Future }
+
+class TsValidationSpec extends PropSpec with PropertyChecks with MustMatchers with TsGenerators with CommonTsValidation with CommonTsValidationSpec {
+
+  override implicit val ec: ExecutionContext = scala.concurrent.ExecutionContext.Implicits.global
+
+  implicit def badTimestamp: Gen[Int] = {
+    Gen.oneOf(0, 1000)
+  }
+
+  property("Transmittal Sheet fails S028 (Timestamp Format)") {
+    forAll(tsGen) { ts =>
+      whenever(ts.id == 1) {
+        val badTs = for {
+          t <- badTimestamp
+          x = ts.copy(timestamp = t)
+        } yield x
+
+        failGenTs(badTs)
```

Comment:
@schbetsy and I noticed that this line doesn't do what we expected.  We thought it was supposed to be testing `TsValidationEngine`, but in fact it calls `CommonTsValidation.validate`.  So it seems that `TsValidationEngine` is not tested at all (even in the JVM version), so `S028` isn't being tested here.